src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -939,28 +940,52 @@ def _prepare_template_uri_with_query_string(template_uri, input_query_string):
         raise InvalidArgumentValueError('Unable to parse parameter: {} .Make sure the value is formed correctly.'.format(input_query_string))
 
 
-def _parse_bicepparam_file(cli_ctx, template_file, parameters):
-    ensure_bicep_installation(cli_ctx)
+def _get_bicepparam_file_path(parameters):
+    bicepparam_file_path = None
 
-    minimum_supported_version = "0.14.85"
-    if not bicep_version_greater_than_or_equal_to(minimum_supported_version):
-        raise ArgumentUsageError(f"Unable to compile .bicepparam file with the current version of Bicep CLI. Please upgrade Bicep CLI to {minimum_supported_version} or later.")
-    if len(parameters) > 1:
-        raise ArgumentUsageError("Can not use --parameters argument more than once when using a .bicepparam file")
-    if template_file and not is_bicep_file(template_file):
-        raise ArgumentUsageError("Only a .bicep template is allowed with a .bicepparam file")
-    bicepparam_file = parameters[0][0]
+    for parameter_list in parameters:
+        for parameter_item in parameter_list:
+            if is_bicepparam_file(parameter_item):
+                if not bicepparam_file_path:
+                    bicepparam_file_path = parameter_item
+                else:
+                    raise ArgumentUsageError("Only one .bicepparam file can be provided with --parameters argument")
+
+    return bicepparam_file_path
+
+
+def _parse_bicepparam_inline_params(parameters, template_param_defs):
+    parsed_inline_params = {}
+
+    for parameter_list in parameters:
+        for parameter_item in parameter_list:
+            if is_bicepparam_file(parameter_item):
+                continue
+
+            if not _try_parse_key_value_object(template_param_defs, parsed_inline_params, parameter_item):
+                raise CLIError(f"Unable to parse parameter: {parameter_item}. Only correctly formatted in-line parameters are allowed with a .bicepparam file")
```

Comment:
```suggestion
                raise InvalidArgumentValueError(f"Unable to parse parameter: {parameter_item}. Only correctly formatted in-line parameters are allowed with a .bicepparam file")
```
How about using the specific error type `InvalidArgumentValueError` instead of `CLIError`?